### PR TITLE
[TASK] Splits GitHub rendering action into composite

### PR DESCRIPTION
**Pull request by garvinhicking**
This PR splits the current process of single-use rendering up into a two-step composite process as outlined in  https://github.com/TYPO3-Documentation/render-guides/issues/158

The first step takes care of configuring a local guides.xml (during rendering/GH-action only) with dynamic parameters. Those parameters fixate the input from the GitHub action for the target GIT branch and target deployment directory.

The second step is the usual rendering, which will then take into account the modified parameters, so that the rendered output includes also the non-committed changes, as the old sphinx jobfile did.

The commit includes a large addition to the `ConfigureCommand` that is needed to configure the required attributes. That code will need later refactoring and added unit testing. For now it should work to progress on the GitHub action.

This PR should be merged in conjunction with the changes made to the `gh-render-action` PR: https://github.com/TYPO3-Documentation/gh-render-action/pull/21

### CODE REVIEW

**Comment by garvinhicking** [comment]
(Addendum: Don't mind the failing "test-self-action" step. It can only succeed once the newly introduced step is merged into `main`.)

**Comment by garvinhicking** [comment]
Another addendum:

Actually I just see that this is currently done in an interesting way via:

```        
run: "sed -i 's/^  image:.*$/  image: \"Dockerfile\"/g' action.yml"
```

in the `workflows/main.yaml` file. I'd need to do the same to the updated action.yml file with the `uses:` attributes. I can do that tomorrow with a fresh mindset, but IMO it shouldn't block merging the PR.

File: action.yml
Comment (brotkrueml) [inline action.yml:20]: Should we really have this as example in the description? This nested structure is not necessary anymore, isn't it?

**Review by linawolf — APPROVED** [review]
Looks good to me as far as I can tell

File: action.yml
Comment (linawolf) [reply action.yml:20]: I think that is the structure that is expected for the rest of the deployment process

Comment (brotkrueml) [reply action.yml:20]: Ah, so once introduced, that's the way ;-) But the default value below is another folder name.

Comment (garvinhicking) [reply action.yml:20]: I actually used this default value due to the action in `gh-render-action` https://github.com/TYPO3-Documentation/gh-render-action/blob/main/action.yml#L62: 

```
    - uses: TYPO3-Documentation/render-guides@main
      id: render-guides
      if: steps.enable_guides.outputs.GUIDES == 'true'
      with:
        working-directory: t3docsproject
        config: ./t3docsproject/Documentation/
        output: ./RenderedDocumentation/Result/project/0.0.0
```

So I wanted to make the example relate to how we actually use it.

Comment (garvinhicking) [reply action.yml:20]: (That path is only used during the CI process, that's also why the version number and so forth is fixed)

Comment (brotkrueml) [reply action.yml:20]: Ack.

**Review by brotkrueml — APPROVED** [review]

**Review by jaapio — COMMENTED** [review]
Just a small remark, apart from that, it looks very good.

File: .github/actions/configure-guides-step/action.yaml
Comment (garvinhicking) [reply .github/actions/configure-guides-step/action.yaml:11]: Thanks, added!

**Pull request merged**
